### PR TITLE
Compute features for GigaSpeech by splitting the manifest.

### PR DESCRIPTION
**Pull request by csukuangfj**
It solves the OOM problem by splitting the manifest before computing features and combining them afterwards.

### CODE REVIEW

File: egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        cut_set = CutSet.from_file(raw_cuts_path)
+
+        logging.info("Computing features")
+
+        cut_set = cut_set.compute_and_store_features_batch(
+            extractor=extractor,
+            storage_path=f"{output_dir}/feats_XL_{idx}",
+            num_workers=args.num_workers,
+            batch_duration=args.batch_duration,
+            storage_type=LilcomHdf5Writer,
```
Comment (wgb14) [inline egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py]: Do we use `storage_type=ChunkedLilcomHdf5Writer` instead if we trim after extraction, as @pzelasko suggested https://github.com/lhotse-speech/lhotse/issues/452#issuecomment-970896841

File: egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            continue
+
+        raw_cuts_path = output_dir / f"cuts_XL_raw.{idx}.jsonl.gz"
+
+        logging.info(f"Loading {raw_cuts_path}")
+        cut_set = CutSet.from_file(raw_cuts_path)
+
+        logging.info("Computing features")
+
+        cut_set = cut_set.compute_and_store_features_batch(
```
Comment (wgb14) [inline egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py:98]: Does `compute_and_store_features_batch` support CPU extraction well? https://github.com/lhotse-speech/lhotse/blob/a6c393602469e96f4255347b6c19fd34b2d2dbe7/lhotse/cut.py#L3768 I assume it's extremely slow.

**Review by wgb14 — APPROVED** [review]
LGTM!
Just left minor comments. I'll merge and test on it.

File: egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py (hunk 2 above)
Comment (csukuangfj) [reply egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py:98]: > Does compute_and_store_features_batch support CPU extraction well?

Yes, `kaldifeat` supports CUDA as well as CPU.

---

> I assume it's extremely slow.

Not extremely slow. Maybe 8x slower than CUDA.

File: egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py (hunk 1 above)
Comment (csukuangfj) [reply egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_splits.py]: Thanks. I did not notice that.

**Comment by wgb14** [comment]
LGTM. Merging this

**Pull request merged**